web/ai_service.py

```
@@ -0,0 +1,267 @@
+
+class AIService:
+    def __init__(self):
+        self.openai_client = None
+        self.gemini_api_key = None
+        self._current_request = None
+        self._request_lock = threading.Lock()
+        self._executor = ThreadPoolExecutor(max_workers=3)
+        self._initialize_services()
+        self._setup_signal_handlers()
+
+    def _setup_signal_handlers(self):
```

Ruff (0.8.2)

Missing return type annotation for private function `_setup_signal_handlers`

Add return type annotation: `None`

(ANN202)